### ISSUE TITLE
"Detours should be called something else

### ISSUE THREAD

**Issue by computron** [materialsproject/jobflow#358]
**Describe the bug**

**To Reproduce**

Provide any example files that are needed to reproduce the error,
especially if the bug pertains to parsing of a file.

**Expected behavior**

**Screenshots**

**Comment by arosen93** [comment]
Related to #341, which needs further clarity because I also found it somewhat confusing.

**Comment by utf** [comment]
I agree that the wording is confusing because of the Fireworks context. That said, the detour does route back to the original workflow but the outputs can't be used. The main difference between detour and addition is the order of execution. A new detour job will always run before any other children are run. Whereas an addition job can get run before/after child jobs depending on the priority of the workflow manager.

<img width="895" alt="detour/addition" src="https://github.com/materialsproject/jobflow/assets/1330638/217879a8-56ac-4020-9004-2dd02eb4ca97">

The original detour in fireworks is more like "replace" in jobflow in the way that's its used.

<img width="428" alt="replace" src="https://github.com/materialsproject/jobflow/assets/1330638/cafc1d08-7d20-4851-9674-60d4f2c1f9d1">

One question is whether we really need both addition and detour. Addition is just a special version of detour where the execution order doesn't matter. We could make the default behaviour of addition the same as detour and remove the detour option entirely. 

If we decide to keep both detour and addition, some options for a new name could be:
- Insert
- Diversion

**Comment by davidwaroquiers** [comment]
Good that this issue is raised and it would indeed be good to clarify. We also had some thinking with @gpetretto about this. And we ended up thinking, if we cannot use the outputs of a detour (which is the case in jobflow), for which case is it useful (and I did not find a case where this would be needed, basically when we need the outputs, we always use replace as it is the "only" option for it) ? So we actually never use detour, only addition or replace.

**Comment by utf** [comment]
I think the actual detour is still useful in contrast to addition. You might want to run some post processing and update a file in place before the next job runs.

**Comment by davidwaroquiers** [comment]
I see, but that means this job needs to run in the directory of the previous one right ? Or at least access the path of the job from which the detour is made and perform it's updates of the said files there.

**Comment by arosen93** [comment]
@utf, I think that would be a great figure to add to the docs!

**Comment by computron** [comment]
This conversation is getting me a bit mixed up

@utf can you put diagrams of what all the current options are in jobflow? The FWs diagram is below:

![image](https://github.com/materialsproject/jobflow/assets/986759/f1bcd717-af7b-4ca6-9d9e-1242256d0bf1)

**Comment by utf** [comment]
Hi @computron, the options available are in this comment: https://github.com/materialsproject/jobflow/issues/358#issuecomment-1613102577

There are also options to `stop_children` and `stop_jobflow` which are the equivalent of `defuse_children` and `exit` in fireworks.

**Comment by utf** [comment]
How about we change `detour` to `insert`?
